awsshell/wizard.py

```diff
@@ -19,6 +21,81 @@
 LOG = logging.getLogger(__name__)
 
 
+class ParamCoercion(object):
+    """This class coerces string parameters into the correct type.
+
+    By default this converts strings to numerical values if the input
+    parameters model indicates that the field should be a number. This is to
+    compensate for the fact that values taken in from prompts will always be
+    strings and avoids having to create specific interactions for simple
+    conversions or having to specify the type in the wizard specification.
+    """
+
+    _DEFAULT_DICT = {
+        'integer': int,
+        'float': float,
+        'double': float,
+        'long': long_type
+    }
+
+    def __init__(self, type_dict=_DEFAULT_DICT):
+        """Initialize a ParamCoercion object.
+
+        :type type_dict: dict
+        :param type_dict: (Optional) A dictionary of converstions. Keys are
+        strings representing the shape type name and the values are callables
+        that given a string will return an instance of an appropriate type for
+        that shape type. Defaults to only coerce numbers.
+        """
+        self._type_dict = type_dict
+
+    def coerce(self, params, shape):
+        """Coerce the params according to the given shape.
+
+        :type params: dict
+        :param params: The parameters to be given to an operation call.
+
+        :type shape: :class:`botocore.model.Shape`
+        :param shape: The input shape for the desired operation.
+
+        :rtype: dict
+        :return: The coerced version of the params.
+        """
+        return self._coerce(params, shape)
+
+    def _coerce(self, params, shape):
+        name = shape.type_name
+        if isinstance(params, dict) and name == 'structure':
+            return self._coerce_structure(params, shape)
+        elif isinstance(params, (list, tuple)) and name == 'list':
+            return self._coerce_list(params, shape)
+        elif isinstance(params, six.string_types) and name in self._type_dict:
+            target_type = self._type_dict[shape.type_name]
+            return self._coerce_field(params, target_type)
+        return params
+
+    def _coerce_structure(self, params, shape):
+        members = shape.members
+        coerced = {}
+        for param in members:
+            if param in params:
+                coerced[param] = self._coerce(params[param], members[param])
+        return coerced
+
+    def _coerce_list(self, param, shape):
+        member_shape = shape.member
+        coerced_list = []
+        for i, item in enumerate(param):
+            coerced_list.append(self._coerce(item, member_shape))
+        return coerced_list
+
+    def _coerce_field(self, value, number_type):
```

Comment:
Ohhhh its not ``number_type`` its suppose to be ``member_type``... That threw me off a bit.